src/app/public/modules/overlay/overlay.service.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ComponentRef,
+  Injectable,
+  Type
+} from '@angular/core';
+
+import {
+  SkyDynamicComponentService
+} from '../dynamic-component';
+
+import {
+  SkyOverlayConfig
+} from './overlay-config';
+
+import {
+  SkyOverlayDomAdapterService
+} from './overlay-dom-adapter.service';
+
+import {
+  SkyOverlayHostComponent
+} from './overlay-host.component';
+
+import {
+  SkyOverlayInstance
+} from './overlay-instance';
+
+/**
+ * This service is used to create new overlays.
+ */
+@Injectable()
+export class SkyOverlayService {
+
+  private host: ComponentRef<SkyOverlayHostComponent>;
+
+  private instances: SkyOverlayInstance<any>[] = [];
+
+  constructor(
+    private dynamicComponentService: SkyDynamicComponentService,
+    private adapter: SkyOverlayDomAdapterService
+  ) {
+    this.createHostComponent();
+  }
+
+  /**
+   * Creates a new overlay and appends an instance of the provided component.
+   * @param component The component to append to the overlay.
+   * @param config The configuration for the overlay.
+   */
+  public create<T>(
+    component: Type<T>,
+    config: SkyOverlayConfig = {
+      closeOnNavigation: true,
+      enableClose: false,
+      enableScroll: true,
+      showBackdrop: false
+    }
```

Comment:
It looks like these defaults aren't getting set. If I deliberately provide values, it works as expected. However, if I just do a config of `{}`, then the defaults don't get set at all.